### PR TITLE
Add one-time schedule

### LINKED ISSUE #133: Add one-time schedule

### PR DESCRIPTION
**Pull request by thu5**
This PR allows users to add one-time schedules for rules. Closes #133.

Screen Shot 2020-09-08 at 2 03 03 PM

### CODE REVIEW

**Review by cathyang-work — COMMENTED** [review]
Can you add a screenshot?

**Comment by thu5** [comment]
Sure

**Review by cathyang-work — APPROVED** [review]

File: lib/src/rule_creator_component.dart [hunk 1]
```diff
@@ -130,7 +130,16 @@ class RuleCreatorComponent {
...
-            ..cronExpression = _cronExpression(month, day, hour, minute));
+            ..cronExpression =
+                _repeatingCronExpression(month, day, hour, minute));
+        break;
+      case Schedule_Type.ONE_TIME:
+        rule.schedule = Schedule()
+          ..type = scheduleTypes[scheduleType]
+          ..timezone = timezone
+          ..oneTimeParams = (Schedule_OneTimeParams()
+            ..cronExpression = '$minute $hour $day $month *'
```
Comment (gg-qinp) [inline lib/src/rule_creator_component.dart:141]: [No Action Required]
1. theoretically we should make sure the time is not a past time,but it is fine for now.
2. current special handling for one time task does not scale for task with more repeating times. Think about how Google Calendar handle repeated tasks, we can store multiple entries with timestamp in a table,  then we can maintain a background job to query the table periodically and trigger the task.

**Review by gg-qinp — APPROVED** [review]

File: lib/src/rule_creator_component.dart (hunk 1 above)
Comment (thu5) [reply lib/src/rule_creator_component.dart:141]: Good points, I'll talk about it in my presentation :) I can handle (1) if I have time though.

**Pull request merged**